model/service_logs/v1/root_resource.model

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright (c) 2019 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Root of the tree of resources of the service logs service.
+resource Root {
+	// Reference to the resource that manages the collection of cluster logs.
+	locator ClusterLogs {
```

Comment:
Do we expect to have other kinds of logs managed by this service? Or only cluster logs?

Reply:
I think we do. but it's better to ask @abhgupta

Reply:
If we expect this service to manage logs for things that may not be clusters then I'd suggest to rename this from `ClusterLogs` to `Entries`. I would also suggest to change the service identifier from `service_logs` to `logs_mgmt`. So the request to get log entries would look like this:

```http
GET /api/logs_mgmt/v1/entries HTTP/1.1
```

The result should then look like this:

```json
{
  "kind": "LogEntryList",
  "page": 1,
  "size": 100,
  "total": 12342,
  "items": [
    {
      "kind": "LogEntry",
      "id": "123",
      "href": "/api/logs_mgmt/v1/entries/123",
      ...
    },
    {
      "kind": "LogEntry",
      "id": "456",
      "href": "/api/logs_mgmt/v1/entries/456",
      ...
    },
    ...
  ]
}
```

In such a generic log management service you will probably want to have in each entry an attribute to indicate the _type_ of the source of that log entry, so you can distinguish (and search) according to that type. For example:

```json
{
  "kind": "LogEntry",
  "id": "123",
  "href": "/api/logs_mgmt/v1/entries/123",
  "source_type": "cluster",
  "source_id": "43baceef-99aa-447b-b02d-35217c9ddcc6",
  "severity": "error",
  ...
}
```

It should then be possible to use those `source_type` and `source_id` to search for the entries for a particular cluster:

```http
GET /api/logs_mgmt/v1/entries?source_type=cluster&source_id=43baceef-99aa-447b-b02d-35217c9ddcc6 HTTP/1.1
```
And it would still be relatively easy to add a new source type (with its own identifiers) without breaking compatibility.

Reply:
@jhernand I apologize, we moved quickly on this service and I don't think I got the design doc over to you before we made the decisions we did. For now we only expect cluster logs, but we do want to prepare for other types of logs in the future that may be completely different objects altogether. 

`/api/logs_mgmt` may have been a better endpoint for this service.